### PR TITLE
fix(storage): resolve cache/storage inconsistencies on HEAD request

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/local.go [hunk 1]
```diff
@@ -1480,6 +1480,16 @@ func (is *ImageStoreLocal) checkCacheBlob(digest string) (string, error) {
 		return "", err
 	}
 
+	if _, err := os.Stat(dstRecord); err != nil {
+		if err := is.cache.DeleteBlob(digest, dstRecord); err != nil {
```
Comment (peusebiu) [inline pkg/storage/local.go:1484]: actually this is not needed for local, because copyBlob() will exit with error later

**Review by andaaron — DISMISSED** [review]
LGTM

**Comment by peusebiu** [comment]
TODO: remove the same check used in other methods

**Comment by rchincha** [comment]
@peusebiu some description about what this bug is fixing?

**Comment by peusebiu** [comment]
> 

it's a small improvement, resolving inconsistency between cache and storage, in case some files are present in cache, but not on storage for some reason.

**Comment by rchincha** [comment]
fix(storage) - resolve cache/storage inconsistencies on HEAD request
^ fix(storage): ...

**Pull request merged**